sdk/identity/azure-identity/CMakeLists.txt

```diff
@@ -12,6 +12,14 @@ project(azure-identity LANGUAGES CXX)
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+# Compile Options
+option(BUILD_SOURCE_DEPS "build source dependencies" OFF)
+
+if(BUILD_SOURCE_DEPS)
+    set(AZ_ALL_LIBRARIES ON)
```

Comment:
this might interfere with a different feature.  We use `AZ_ALL_LIBRARIES` as a sentinel that the cmake generate was invoked from the root path.  Why are you setting it to ON here?

I am worry re-using the same sentinel with a different meaning will lead us to bugs.  Can we introduce a new variable that is used only for build-from-source feature?

Reply:
the thing here is that azure core and identity use this flag internally, since i'm using it from older sources versions than the latest and greatest , i cannot make changes to the flag used. 
This flag is used ony for deciding to include the GTest framework. 
And in the end the flag is used as intended , i am building ALL AZ LIBRARIES. If you think about it i am building the same way as it building from root with the only difference is that i point to diffrent dependancies source code than the ones in the main branch.

Reply:
> with the only difference is that i point to diffrent dependancies source code than the ones in the main branch.

That difference is huge.  Right now, if I run CMakeList from the root, the option ALL_AZ_LIB... will be set to ON. I think it is not good to use the same option for something other than that.